### ISSUE TITLE
Over engineered, slow-as-hell build times, exaggeratedly complex Redux logic

### ISSUE THREAD

**Issue by webpolis** [infinitered/ignite#1247]
Yeah, this is what I have to say. Is there any reason you made it so complex in design?

Every new build is a hell of a problem, and I spend like ... 20 minutes on each build.

I bet there are less much complex and faster projects out there. Anyone, please advise or provide alternatives.

**Comment by GantMan** [comment]
always happy to make things simpler.

What do you recommend?

We're actually about to release a new boilerplate using mobx-state-tree.  Let me know if you want to see that one.

**Comment by webpolis** [comment]
Why not using an observable strategy, with RXJs?

**Comment by webpolis** [comment]
I have implemented a Hook based approach using RxJS, at it has made my life so much easy, I can quickly interact from any place within the whole life cycle of my application state.

**Comment by GantMan** [comment]
Sounds like you would love mobx over redux anyhow.  

We're moving towards using something similar, but we really like the state-tree style of events.   So we're going to use mobx-state-tree which is a good mix.  Gives that hook functionality with observers.

We'll see, it might be great!

**Comment by webpolis** [comment]
Sounds great, I have started to hate the whole Redux ecosystem; it just adds redundant complexity with all that reducers, actions and store hell. It makes so hard to follow any simple logic, and you have to maintain 3 different files just to increment a counter. Too over engineered, not my type at all.

**Comment by GantMan** [comment]
It was def better than Flux, but yeah it's still evolving.  I feel like we're still not there yet.  GraphQL + state management is getting to where we all want to be.

**Comment by Amnesthesia** [comment]
I've replaced the sagas + redux with Apollo and GraphQL, and I've managed to cut down the javascript code by literally 70% or something, and the performance of the app is through the roof. 

I don't think I will be using ignite with redux and redux-sagas again, it doesn't seem too well suited for mobile applications that need a lot of data unfortunately.

I'm crossing my fingers for an Apollo / GraphQL boilerplate

**Comment by thebigredgeek** [comment]
I find immense irony in the idea that Rx is simpler than Sagas.  Async/await + array operators.  You don't need Rx.
